src/mono/mono/metadata/appdomain.c

```diff
@@ -1256,27 +1263,62 @@ mono_runtime_install_appctx_properties (void)
 {
 	ERROR_DECL (error);
 	gpointer args [3];
+	int n_total_props;
+	char **total_keys_raw;
+	char **total_values_raw;
+	gunichar2 **total_keys;
+	gunichar2 **total_values;
 
 	MonoMethod *setup = mono_class_get_method_from_name_checked (mono_class_get_appctx_class (), "Setup", 3, 0, error);
 	g_assert (setup);
 
 	// FIXME: TRUSTED_PLATFORM_ASSEMBLIES is very large
 
+	// Combine and convert properties
+	n_total_props = n_appctx_props + n_runtimeconfig_json_props;
+	total_keys_raw = g_new0 (char*, n_total_props);
+	total_values_raw = g_new0 (char*, n_total_props);;
+	total_keys = g_new0 (gunichar2*, n_total_props);
+	total_values = g_new0 (gunichar2*, n_total_props);
+	for (int i = 0; i < n_appctx_props; ++i) {
+		total_keys_raw [i] = g_new0 (char, strlen (appctx_keys [i]));
+		total_values_raw [i] = (char *) malloc (sizeof (char) * (strlen (appctx_values [i]))); // g_new0 doesn't work well with TRUSTED_PLATFORM_ASSEMBLIES's value
```

Comment:
??? That doesn't sound right.  `g_new0` shouldn't care

Reply:
This was probably because the memory has already been deallocated.